creator-node/src/utils.js

```diff
@@ -77,9 +82,13 @@ const wait = (ms) => new Promise((resolve, reject) => setTimeout(() => reject(ne
 /** Cat single byte of file at given filepath. */
 const ipfsSingleByteCat = (path, req) => new Promise(async (resolve, reject) => {
   const start = Date.now()
-  let ipfs = req.app.get('ipfsAPI')
+  let ipfs = req.app.get('ipfsLatestAPI')
+
   try {
-    await ipfs.cat(path, { length: 1 })
+    const chunks = []
+    for await (const chunk of ipfs.cat(path, { length: 1, timeout: 1000 })) {
```

Comment:
why do we need to iterate on every chunk here? shouldn't the first be enough? not necessarily blocking, jw

Reply:
so async iterator doesn't return data at a given rate from ipfs. this data format is just a stream of data backed by a generator that sends data at some rate. so basically the input params like length are what we want from the generator and the chunks is the output. it could resolve in one chunk or take several, no way of knowing how many chunks is enough

Reply:
the new api has an example of the cat call using a for-loop https://github.com/ipfs/interface-js-ipfs-core/blob/master/SPEC/FILES.md#cat, but yes, this for-loop shouldn't be necessary.

will check what chunk consists of and update as necessary

Reply:
@dmanjunath right but if 1 of the chunks succeed, the rest don't matter - that's what i was getting at. it might be >1, but the subsequent chunks are irrelevant to us since the multihash is available